src/sage/combinat/permutation.py

```diff
@@ -1242,7 +1244,23 @@ def to_alternating_sign_matrix(self):
         from sage.combinat.alternating_sign_matrix import AlternatingSignMatrix
         return AlternatingSignMatrix(self.to_matrix().rows())
 
-    def __mul__(self, rp) -> Permutation:
+    def __mul__(self, rp):
```

Comment:
why having both `__mul__` and `_mul_` ?

Reply:
Same reason we have them for other elements: When we know the object is in the correct parent, we can avoid the call through the coercion framework. Plus we need `_mul_()` for `prod`.